frontend/src/pages/hardwareProfiles/utils.ts

```diff
@@ -1,4 +1,224 @@
 import { HardwareProfileKind } from '~/k8sTypes';
+import { Identifier, IdentifierResourceType } from '~/types';
+import { determineUnit, splitValueUnit } from '~/utilities/valueUnits';
+import { hasCPUandMemory } from './manage/ManageNodeResourceSection';
+
+export type WarningNotification = {
+  warningStatus: boolean;
+  title: string;
+  message: string;
+  isDefaultProfile: boolean;
+};
+
+export type HardwareProfileWarningMessage = {
+  warning: boolean;
+  title: string;
+  message: string;
+  isEnabled: boolean;
+};
 
 export const isHardwareProfileOOTB = (hardwareProfile: HardwareProfileKind): boolean =>
   hardwareProfile.metadata.labels?.['opendatahub.io/ootb'] === 'true';
+
+export const defaultNodes: Identifier[] = [
+  {
+    displayName: 'CPU',
+    identifier: 'cpu',
+    minCount: 1,
+    maxCount: 4,
+    defaultCount: 2,
+    resourceType: IdentifierResourceType.CPU,
+  },
+  {
+    displayName: 'Memory',
+    identifier: 'memory',
+    minCount: '2Gi',
+    maxCount: '8Gi',
+    defaultCount: '4Gi',
+    resourceType: IdentifierResourceType.MEMORY,
+  },
+];
+
+export const generateWarningTitle = (
+  isEnabled: boolean,
+  isAllWarnings: boolean,
+  warningCount: number,
+): string => {
+  if (!isEnabled) {
+    return 'All hardware profiles are disabled';
+  }
+  if (isAllWarnings) {
+    return 'All hardware profiles are invalid';
+  }
+  if (warningCount > 0) {
+    return 'One or more hardware profiles are invalid';
+  }
+  return '';
+};
+
+export const generateWarningMessage = (
+  isEnabled: boolean,
+  isAllWarnings: boolean,
+  warningCount: number,
```

Comment:
You don't need warningCount here. There is no way warningCount could be `0` in this situation when showing the alert.